src/index.ts

```diff
@@ -27,18 +27,22 @@ export function newConfig<S>(spec: S, env: Environment, options: ConfigOptions =
   const errors: Error[] = [];
   // go through our spec version of S that the type system thinks has primitives
   // but are really ConfigOptions that we've casted to the primitive
-  const config = {};
-  Object.keys(spec).forEach(k => {
+  const config: Record<string, S | unknown> = {};
+  Object.keys(spec).forEach((k) => {
     try {
-      const v = (spec as any)[k];
+      const v = (spec as Record<string, unknown>)[k];
       if (v instanceof ConfigOption) {
-        (config as any)[k] = v.getValue(k, env, options);
+        config[k] = v.getValue(k, env, options);
       } else if (typeof v === "object") {
         // assume this is a nested config spec
-        (config as any)[k] = newConfig(v, env, options);
+        config[k] = newConfig(v, env, options);
       }
     } catch (e) {
-      errors.push(e);
+      if (e instanceof Error) {
+        errors.push(e);
+      } else {
+        throw e;
+      }
```

Comment:
This is because newer TypeScript treats errors as `unknown`. I figure it's safest just to re-throw if it's not an `Error`.